### PR TITLE
Refactor syndicate buylist

### PR DESCRIPTION
**Pull request by Glamurio**
[Internal] [Code Quality] [QoL] [Secret]

## About the PR 
Changes `/datum/syndicate_buylist/item`, `/datum/syndicate_buylist/item2` and `/datum/syndicate_buylist/item3` to `/datum/syndicate_buylist/items`, a list.

Refactors `/datum/syndicate_buylist/run_on_spawn()` to put items purchased into the hand, rather than on the floor, except for surplus crate contents.

Changes any behavior that had to do with  `/datum/syndicate_buylist/item` to instead iterate over the list.

## Why's this needed? 
Having purchases contain a list of items is cleaner instead of having them be hardcoded values. Spawning items in your hand is just QoL in line with many other systems in the game.

### CODE REVIEW

**Comment by Glamurio** [comment]
Looks like the chicken grenades will need a secret fix.

File: code/obj/storage/crates.dm [hunk 1]
```diff
@@ -338,14 +338,16 @@
...
+				if(length(item_datum.items) == 0) continue
+				for (var/item in item_datum.items)
+					var/obj/item/I = new item(src)
+					I.Scale(NESTED_SCALING_FACTOR**nest_amt, NESTED_SCALING_FACTOR**nest_amt) //scale the contents if we're nested
+					if (owner)
+						item_datum.run_on_spawn(I, owner, TRUE, owner_uplink)
+						var/datum/antagonist/traitor/T = owner.mind?.get_antagonist(ROLE_TRAITOR)
+						if (istype(T))
+							T.surplus_crate_items.Add(item_datum)
+					telecrystals += item_datum.cost
```
Comment (Tarmunora) [inline code/obj/storage/crates.dm:350]: Indentation error - looks like this should be outside the `for item in items` loop.

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/storage/crates.dm (hunk 1 above)
Comment (Glamurio) [reply code/obj/storage/crates.dm]: Ooops, looks like a faulty merge

File: code/datums/syndicate_buylist.dm [hunk 2]
```diff
@@ -18,12 +18,8 @@ ABSTRACT_TYPE(/datum/syndicate_buylist)
...
 	/// Name of the buylist entry
 	var/name = null
-	/// A typepath for the item that will be spawned when the datum is purchased
-	var/atom/item = null
-	/// A typepath for the second, optional, item that will be spawned when the datum is purchased
-	var/atom/item2 = null
-	/// A typepath for the third, optional, item that will be spawned when the datum is purchased
-	var/atom/item3 = null
+	/// A typepath for the items that will be spawned when the datum is purchased
+	var/list/atom/items = list()
```
Comment (ZeWaka) [inline code/datums/syndicate_buylist.dm:22]:
```suggestion
	/// Typepaths of the items that will be spawned when the datum is purchased
	var/list/atom/items = list()
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**